src/main/java/de/chojo/repbot/listener/SelfCleanupService.java

```diff
@@ -0,0 +1,110 @@
+        var selfCleanupService = new SelfCleanupService(shardManager, localizer, dataSource, configuration);
+        service.scheduleAtFixedRate(selfCleanupService, 1, 60, TimeUnit.MINUTES);
+        return selfCleanupService;
+    }
+
+
+    @Override
+    public void run() {
+        if (!configuration.selfCleanup().isActive()) return;
+
+        for (var guild : shardManager.getGuilds()) {
+            var optGuildSettings = guildData.getGuildSettings(guild);
+            if (optGuildSettings.isEmpty()) continue;
+            var settings = optGuildSettings.get();
+            if (settings.activeChannel().isEmpty() && settings.isChannelWhitelist()) {
+                promptCleanup(guild);
+                continue;
+            }
+            guildData.cleanupDone(guild);
+        }
+
+        for (var guildId : guildData.getCleanupList()) {
+            var guild = shardManager.getGuildById(guildId);
+            if (guild != null) notifyCleanup(guild);
+        }
+    }
+
+    private void promptCleanup(Guild guild) {
+        var selfMember = guild.getSelfMember();
+        if (selfMember.getTimeJoined().isAfter(configuration.selfCleanup().getPromptDaysOffset())) return;
+        if (configuration.botlist().isBotlistGuild(guild.getOwnerIdLong())) return;
+        if (guildData.getCleanupPromptTime(guild).isPresent()) return;
+        guildData.selfCleanupPrompt(guild);
+
+        var embed = new LocalizedEmbedBuilder(localizer, guild)
+                .setTitle("selfCleanup.prompt.title")
+                .setDescription(localizer.localize("selfCleanup.prompt", guild,
+                        Replacement.create("DAYS", configuration.selfCleanup().leaveDays())))
+                .build();
+
+        notifyGuild(guild, embed);
+        log.info(LogNotify.STATUS, "Promptet guild self cleanup.");
+    }
+
+    private void notifyCleanup(Guild guild) {
+        if (guildData.getCleanupPromptTime(guild).get().isAfter(configuration.selfCleanup().getLeaveDaysOffset()))
+            return;
+
+        var embed = new LocalizedEmbedBuilder(localizer, guild)
+                .setTitle("selfCleanup.leave.title")
+                .setDescription(localizer.localize("selfCleanup.leave", guild,
+                        Replacement.create("INVITE", configuration.links().invite())))
+                .build();
+
+        notifyGuild(guild, embed);
+        guild.leave().queue();
+        log.info(LogNotify.STATUS, "Left guild caused by self cleanup.");
+    }
+
+
+    private void notifyGuild(Guild guild, MessageEmbed embed) {
+        var selfMember = guild.getSelfMember();
+        guild.retrieveMemberById(guild.getOwnerIdLong())
+                .flatMap(member -> member.getUser().openPrivateChannel())
+                .flatMap(privateChannel -> privateChannel.sendMessageEmbeds(embed))
+                .complete();
```

Comment:
This will cause an exception if the owner has direct messages disabled so the channel notify would not work in that case

Reply:
I added error handling